From 7900cfc98f221396a3c7b922dc4b51bb0961ae68 Mon Sep 17 00:00:00 2001
From: Damjan Jovanovic <damjan@apache.org>
Date: Sun, 19 Oct 2014 07:03:08 +0000
Subject: [PATCH] Fixed "ExifReWriter always writes EXIF segment before JFIF
 segment."

Jira issue key: IMAGING-140
Submitted by: Gavin Shiels <shecks at gmail dot com>



git-svn-id: https://svn.apache.org/repos/asf/commons/proper/imaging/trunk@1632877 13f79535-47bb-0310-9956-ffa450edef68
---
 pom.xml                                                        | 3 +++
 src/changes/changes.xml                                        | 3 +++
 .../apache/commons/imaging/formats/jpeg/exif/ExifRewriter.java | 2 +-
 3 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/pom.xml b/pom.xml
index 27f011fdc..1828ab343 100644
--- a/pom.xml
+++ b/pom.xml
@@ -354,6 +354,9 @@
     <contributor>
       <name>Gary Lucas</name>
     </contributor>
+    <contributor>
+      <name>Gavin Shiels</name>
+    </contributor>
     <contributor>
       <name>Peter Royal</name>
     </contributor>
diff --git a/src/changes/changes.xml b/src/changes/changes.xml
index 41ef4e8bf..83ce866e5 100644
--- a/src/changes/changes.xml
+++ b/src/changes/changes.xml
@@ -46,6 +46,9 @@ The <action> type attribute can be add,update,fix,remove.
   <body>
 
     <release version="1.0" date="TBA" description="TBA">
+      <action issue="IMAGING-140" dev="damjan" type="fix" due-to="Gavin Shiels">
+        ExifReWriter always writes EXIF segment before JFIF segment.
+      </action>
       <action issue="IMAGING-131" dev="damjan" type="fix">
         Allow null parameters in PngImageParser.getBufferedImage(), and add some tests for null parameters.
       </action>
diff --git a/src/main/java/org/apache/commons/imaging/formats/jpeg/exif/ExifRewriter.java b/src/main/java/org/apache/commons/imaging/formats/jpeg/exif/ExifRewriter.java
index a38845e98..4a6848512 100644
--- a/src/main/java/org/apache/commons/imaging/formats/jpeg/exif/ExifRewriter.java
+++ b/src/main/java/org/apache/commons/imaging/formats/jpeg/exif/ExifRewriter.java
@@ -525,7 +525,7 @@ private void writeSegmentsReplacingExif(final OutputStream os,
                 if (firstSegment.marker == JpegConstants.JFIF_MARKER) {
                     index = 1;
                 }
-                segments.add(0, new JFIFPieceSegmentExif(JpegConstants.JPEG_APP1_MARKER,
+                segments.add(index, new JFIFPieceSegmentExif(JpegConstants.JPEG_APP1_MARKER,
                         markerBytes, markerLengthBytes, newBytes));
             }
 
